src/Symfony/Component/DependencyInjection/Loader/IniFileLoader.php

```diff
@@ -52,7 +52,15 @@ public function load($resource, $type = null)
      */
     public function supports($resource, $type = null)
     {
-        return is_string($resource) && 'ini' === pathinfo($resource, PATHINFO_EXTENSION);
+        if (!is_string($resource)) {
+            return false;
+        }
+
+        if (null === $type && 'ini' === pathinfo($resource, PATHINFO_EXTENSION)) {
+            return true;
+        }
+
+        return null !== $type && 'ini' === $type;
```

Comment:
@javiereguiluz the extension is checked only when the type is null. So there is no question of priority